File: Frontend/src/models/Employee.ts
import BaseResponse from "./BaseResponse";
import { DepartmentResponse } from "./Department";
import { JobLevelResponse } from "./JobLevel";
import { JobTitleResponse } from "./JobTitle";
import { JobTypeResponse } from "./JobType";
import { OfficeResponse } from "./Office";
import { UserResponse, UserRequest } from "./User";

export interface EmployeeResponse extends BaseResponse {
    user: UserResponse;
    office: OfficeResponse;
    department: DepartmentResponse;
    jobType: JobTypeResponse;
    jobLevel: JobLevelResponse;
    jobTitle: JobTitleResponse;
}

export interface EmployeeRequest {
    user: UserRequest;
    officeId: number;
    departmentId: number;
    jobTypeId: number;
    jobLevelId: number;
    jobTitleId: number;
}

File: Backend/src/main/java/backend/osiris/entity/employee/Employee.java
package backend.osiris.entity.employee;

import backend.osiris.entity.BaseEntity;
import backend.osiris.entity.authentication.User;
import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Accessors(chain = true)
@Entity
@Table(name = "employee")
public class Employee extends BaseEntity {
    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "user_id", referencedColumnName = "id", nullable = false, unique = true)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "office_id", nullable = false)
    @JsonBackReference
    private Office office;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "department_id", nullable = false)
    @JsonBackReference
    private Department department;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "job_type_id", nullable = false)
    @JsonBackReference
    private JobType jobType;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "job_level_id", nullable = false)
    @JsonBackReference
    private JobLevel jobLevel;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "job_title_id", nullable = false)
    @JsonBackReference
    private JobTitle jobTitle;
}